client/src/js/Matrix.js

```diff
@@ -1,51 +1,62 @@
-// https://github.com/CodingTrain/Toy-Neural-Network-JS/blob/master/lib/matrix.js
-
+ /**
+ * Checkout Chapter 10 of Nature of Code by Daniel "Coding Train" Shiffman, on
+ * Neural Networks. https://natureofcode.com/book/chapter-10-neural-networks/
+ * https://github.com/CodingTrain/Toy-Neural-Network-JS/blob/master/lib/matrix.js
+ */
 class Matrix {
+    /**
+     * @param {number} rows
+     * @param {number} cols
+     */
     constructor(rows, cols) {
         this.rows = rows;
         this.cols = cols;
         this.data = Array(this.rows).fill().map(() => Array(this.cols).fill(0));
     }
 
+    /** Return a new matrix with copied values */
     copy() {
         let m = new Matrix(this.rows, this.cols);
-        for (let i = 0; i < this.rows; i++) {
-            for (let j = 0; j < this.cols; j++) {
-                m.data[i][j] = this.data[i][j];
-            }
-        }
+        m.data = this.data.map(row => row.map(cell => cell));
```

Comment:
A shorthand way of writing it like this.
```
m.data = this.data.map(function(row) => {
    return row.map(function(cell) => {
        return cell;
    }
});
```
 I'll explain all the syntax (and shorthand syntax) of anonymous functions next time we call.